### PR TITLE
Update filename generation for csv files

### PR DESCRIPTION
**Pull request by nshumoogum**
### What

Full dataset downloads and filtered downloads will have a new location and filename when stored in aws.

This is to make sure the filenames are human readable.

### How to review

- Check code changes make sense
- Unit tests pass
- Generate a new filter on cmd dataset and successfully download csv and csvw (metadata file)
- Login as a florence user and create a new instance of a dataset, publish this dataset and check you can download the new version (full dataset) csv and csvw via the website

The new filename should be like so:

- Full dataset - `{dataset-id}-{edition}-v{version}.csv`
- Filtered dataset file - `{dataset-id}-{edition}-v{version}-filtered-{ISO-Date}.csv`

Depends on https://github.com/ONSdigital/dp-filter-api/pull/161

### Who can review

Anyone

### CODE REVIEW

File: event/handler.go [hunk 1]
```diff
@@ -289,10 +298,10 @@ func (handler *ExportHandler) fullDownload(ctx context.Context, event *FilterSub
 		"version":     event.Version,
 	})
 
-	fileID := uuid.NewV4().String()
-	log.Event(ctx, "storing pre-publish file", log.INFO, log.Data{"fileID": fileID})
+	name := event.DatasetID + "-" + event.Edition + "-v" + event.Version
```
Comment (DavidSubiros) [inline event/handler.go:301]: Is it possible that we want to generate more than one file with the same datasetID, Edition and Version? In this case we would have a name collision. 
I imagine that files will be unique by {DatasetID, Edition, Version}, but just wanted to double check.

**Review by DavidSubiros — APPROVED** [review]
approved - but I just added a comment (it's more of a question)

File: event/handler.go (hunk 1 above)
Comment (nshumoogum) [reply event/handler.go:301]: I like the question, but no there shouldn't be a name clash as the full dataset should only be run once. However if there was an issue in processing and the file was uploaded to s3 but the version document failed to update the status to `published` then we may be backed into a corner (e.g. if this function is not idempotent)?

I will double check the `handler.fileStore.PutFile` and check that it does overwrite an existing file if the name does clash

Comment (nshumoogum) [reply event/handler.go:301]: I've double checked and the s3 function we use to upload file to s3 does overwrite an existing file that has the same file location and name. Tested this by hardcoding the name of the file in both dp-dataset-exporter and dp-dataset-exporter-xlsx services.

Im still happy with this!

Comment (DavidSubiros) [reply event/handler.go:301]: Thanks! 👍

**Pull request merged**